spyder/utils/conda.py

```diff
@@ -114,12 +84,21 @@ def get_conda_env_path(pyexec, quote=False):
 
 def find_conda():
     """Find conda executable."""
-    # First try the environment variables
-    conda = os.environ.get('CONDA_EXE') or os.environ.get('MAMBA_EXE')
+    conda = None
+
+    # First try Spyder's conda executable
+    if is_conda_based_app():
+        conda = osp.join(osp.dirname(os.environ['CONDA_EXE']), 'mamba')
```

Comment:
In the Windows case, I think this needs to be `mamba.exe` otherwise validations related to the path (like if it is a file or it exists) done elsewhere using the value returned by this function can fail. Noticed this while updating the Spyder-env-manager plugin which does a compatibility check and it was failing with the message about the conda executable not being available. The code validating this uses `pathlib.Path.exists` which returns False unless you add the extension of the file: https://github.com/spyder-ide/spyder-env-manager/blob/37f2837a077bd5dabd2b123c015c5f071826e8b9/spyder_env_manager/spyder/plugin.py#L111-L113